### PR TITLE
Add --subm-catalog-update flag to run.sh

### PR DESCRIPTION
**Pull request by NogaTal98**
- Add a function that update the submariner catalog source on managed clusters
- Add --subm-catalog-update flag to the the run.sh
- Add this argument to the documentation

### CODE REVIEW

File: lib/submariner_prepare/downstream_prepare.sh [hunk 1]
```diff
@@ -114,6 +114,26 @@ function create_catalog_source() {
...
+    local submariner_version="$SUBMARINER_VERSION_INSTALL"
+    submariner_version=${submariner_version%.*}
+    submariner_version=$(bc <<< "$submariner_version+0.01")
+    submariner_version=$(printf "%.2f" "$submariner_version")
+    export SUBMARINER_VERSION_INSTALL=$submariner_version
+
+    INFO "Update catalog source on the managed clusters for submariner version - $SUBMARINER_VERSION_INSTALL"
+    for cluster in $MANAGED_CLUSTERS; do
+        get_latest_iib
+        image_source="$LATEST_IIB"
```
Comment (MaxBab) [inline lib/submariner_prepare/downstream_prepare.sh:128]: You're actually making identical versions assignment here, in line 128 and in line 131.
Please, remove one of them.

File: lib/submariner_prepare/downstream_prepare.sh [hunk 2]
```diff
@@ -114,6 +114,26 @@ function create_catalog_source() {
...
+
+        INFO "Update catalog source on $cluster cluster"
+        IMG_SRC="$image_source" 
+        KUBECONFIG="$KCONF/$cluster-kubeconfig.yaml" oc -n openshift-marketplace patch catalogsource submariner-catalog --type=merge -p '{"spec":{"image":"'"$IMG_SRC"'"}}'
```
Comment (MaxBab) [inline lib/submariner_prepare/downstream_prepare.sh:132]: The line is very long.
You could break it to a number of lines:

```
    KUBECONFIG="$KCONF/$cluster-kubeconfig.yaml" \
        oc -n openshift-marketplace patch catalogsource \
        submariner-catalog --type=merge -p '{"spec":{"image":"'"$IMG_SRC"'"}}'
```

File: run.sh [hunk 3]
```diff
@@ -135,6 +135,11 @@ function report() {
     report_polarion
 }
 
+function upgrade_submariner() {
```
Comment (MaxBab) [inline run.sh:138]: The name could mislead.
We are not upgrading the submariner.
We are updating the catalog source.

You can change the function name to something like: update_subm_catalog or update_submariner_catalog

File: run.sh [hunk 4]
```diff
@@ -157,6 +162,10 @@ function parse_arguments() {
                 RUN_COMMAND="deploy"
                 shift
                 ;;
+            --upgrade)
```
Comment (MaxBab) [inline run.sh:165]: As above, the name of misleading.
We are not upgrading the submariner.

Please, change it to something like: --subm-catalog-update

**Comment by MaxBab** [comment]
Please, change the name of the commit according to the new argument name.
And add this argument to the documentation.

File: README.md [hunk 5]
```diff
@@ -64,6 +64,8 @@ export OC_CLUSTER_PASS=<password of the cluster user>
 
     --test                 - Perform testing of the Submariner addon
 
+    --subm-catalog-update  - Perform update of the Submariner's catalog source
```
Comment (MaxBab) [inline README.md:67]: Sorry for nagging.

Could you add to the description:

```
Update Submariner's catalog source to the next version
```

**Review by MaxBab — APPROVED** [review]

**Comment by MaxBab** [comment]
Ah, merged by mistake. :)
Never mind.
The last comment was not critical.

**Comment by MaxBab** [comment]
Please, backport it to the older releases as well.

**Comment by NogaTal98** [comment]
/release-2.6

**Comment by NogaTal98** [comment]
/release-2.7

**Pull request merged**